vql/server/repository.go

```diff
@@ -95,6 +95,88 @@ func (self ArtifactSetFunction) Info(
 	}
 }
 
+type ArtifactDeleteFunctionArgs struct {
+	Name string `vfilter:"optional,field=name,doc=The Artifact to delete"`
+}
+
+type ArtifactDeleteFunction struct{}
+
+func (self *ArtifactDeleteFunction) Call(ctx context.Context,
+	scope *vfilter.Scope,
+	args *ordereddict.Dict) vfilter.Any {
+
+	arg := &ArtifactDeleteFunctionArgs{}
+	err := vfilter.ExtractArgs(scope, args, arg)
+	if err != nil {
+		scope.Log("artifact_delete: %v", err)
+		return vfilter.Null{}
+	}
+
+	config_obj, ok := artifacts.GetServerConfig(scope)
+	if !ok {
+		scope.Log("artifact_delete: Command can only run on the server")
+		return vfilter.Null{}
+	}
+
+	manager, _ := services.GetRepositoryManager()
+	if manager == nil {
+		scope.Log("artifact_delete: Command can only run on the server")
+		return vfilter.Null{}
+	}
+
+	global_repository, err := manager.GetGlobalRepository(config_obj)
+	if err != nil {
+		scope.Log("artifact_delete: %v", err)
+		return vfilter.Null{}
+	}
+
+	definition, pres := global_repository.Get(config_obj, arg.Name)
+	if !pres {
+		scope.Log("artifact_delete: Artifact %v not found", arg.Name)
+		return vfilter.Null{}
+	}
+
+	// Same criteria as
+	// https://github.com/Velocidex/velociraptor/blob/3eddb529a0059a05c0a6c2c7057446f36c4e9a6a/gui/static/angular-components/artifact/artifact-viewer-directive.js#L62
+	if !strings.HasPrefix(definition.Name, "Custom.") {
+		scope.Log("artifact_delete: Can only delete custom artifacts.")
+		return vfilter.Null{}
+	}
+
+	var permission acls.ACL_PERMISSION
```

Comment:
Thanks for this. This is the correct check but I wonder if it would be a little higher. As it is currently, one could check if an artifact exists with no permissions.

Reply:
What would the default permission be? I guess I can do to check if the user can write any artifacts at all before getting the permission of the artifact itself.

```
has_no_server_write = vql_subsystem.CheckAccess(scope, acls.SERVER_ARTIFACT_WRITER)
has_no_client_write = vql_subsystem.CheckAccess(scope, acls.ARTIFACT_WRITER)
if (has_no_server_write == nil && has_client_write == nil){ 
     return vfilter.Null{}
}
```

Reply:
Oh i see what you mean. Yeah we definitely need to find the artifact first so we can determine the required permission for it. This would make the check in the correct place then and there behavior is the same as the other artifacts.